launcher/LaunchController.cpp

```diff
@@ -187,16 +187,22 @@ void LaunchController::login()
                         message = tr("Choose your demo mode player name.");
                     }
 
-                    QString lastOfflinePlayerName = APPLICATION->settings()->get("LastOfflinePlayerName").toString();
-                    QString usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
-                    QString name = QInputDialog::getText(m_parentWidget, tr("Player name"), message, QLineEdit::Normal, usedname, &ok);
-                    if (!ok) {
-                        tryagain = false;
-                        break;
+                    QString usedname;
+                    if(m_offlineName.isEmpty()) {
+                        QString lastOfflinePlayerName = APPLICATION->settings()->get("LastOfflinePlayerName").toString();
+                        QString usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
```

Comment:
```suggestion
                        usedname = lastOfflinePlayerName.isEmpty() ? m_session->player_name : lastOfflinePlayerName;
```

Reply:
Great catch! Thank you! Okay, new changes pushed.